### PR TITLE
new features for 0.4.0

### PR DESCRIPTION
**Pull request by guimarqu**
Do not squash.

### CODE REVIEW

**Comment by rrsadykov** [comment]
We should merge master to this branch, no?

File: src/Algorithm/colgen.jl [hunk 1]
```diff
@@ -255,7 +256,8 @@ function compute_red_cost(
     else
         red_cost = getvalue(spsol)
     end
-    red_cost -= (spinfo.lb * spinfo.lb_dual + spinfo.ub * spinfo.ub_dual)
```
Comment (guimarqu) [inline src/Algorithm/colgen.jl:258]: This is causing a bug in one of my experiments, so I'm going to merge this branch into master & tag a patch very soon.

Comment (rrsadykov) [reply src/Algorithm/colgen.jl:258]: I do not understand. I have already replaced this line by 

`red_cost -= (spinfo.lb_dual + spinfo.ub_dual)`

to correct the calculation of the reduced cost

Comment (guimarqu) [reply src/Algorithm/colgen.jl:258]: Yes and I'm going to merge it.

**Review by rrsadykov — APPROVED** [review]

**Pull request merged**